src/test/regress/sql/multi_with_basics.sql

```diff
@@ -0,0 +1,201 @@
+-- CTE cannot be outside of FROM/WHERE clause
+WITH cte_basic AS (
+	SELECT * FROM users_table
+)
+SELECT cte_basic, user_id FROM users_table;
```

Comment:
I think the more relevant example is `SELECT (SELECT user_id FROM cte_basic), user_id FROM users_table` since this is something that would work in postgres, but not in Citus.